src/Core/Auth/UserFeatures/TwoFactorAuth/TwoFactorIsEnabledQuery.cs

```diff
@@ -65,19 +90,146 @@ await TwoFactorEnabledAsync(userDetail.GetTwoFactorProviders(),
 
     public async Task<bool> TwoFactorIsEnabledAsync(ITwoFactorProvidersUser user)
     {
-        var userId = user.GetUserId();
-        if (!userId.HasValue)
+        if (_featureService.IsEnabled(FeatureFlagKeys.PremiumAccessQuery))
+        {
+            var userId = user.GetUserId();
+            if (!userId.HasValue)
+            {
+                return false;
+            }
+
+            var userEntity = user as User ?? await _userRepository.GetByIdAsync(userId.Value);
+            if (userEntity == null)
+            {
+                return false;
```

Comment:
I think this should throw - there should be a valid `User` in the database.

Reply:
Agreed! Updated.